From 018d902e1e6574fdbea48a2a4216102d6e05fca4 Mon Sep 17 00:00:00 2001
From: Giordon Stark <kratsg@gmail.com>
Date: Wed, 25 Feb 2026 06:19:56 +0100
Subject: [PATCH] [PATCH] strip debug for osx

---
 cmake/SupportFunctionsLibrary.cmake | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/cmake/SupportFunctionsLibrary.cmake b/cmake/SupportFunctionsLibrary.cmake
index 3ac484ac..e13877b2 100644
--- a/cmake/SupportFunctionsLibrary.cmake
+++ b/cmake/SupportFunctionsLibrary.cmake
@@ -63,6 +63,14 @@ function(detect_in_tdaq_install OUT_VAR)
     endif()
 endfunction()
 
+if(APPLE)
+  set(STRIP_DEBUG_OPTION -S)
+  set(STRIP_UNNEEDED_OPTION "")
+else()
+  set(STRIP_DEBUG_OPTION --strip-debug)
+  set(STRIP_UNNEEDED_OPTION --strip-unneeded)
+endif()
+
 # debug symbol stripping ---------------------------------------------------------------------------------------
 function(post_build_debug_library name)
   detect_in_tdaq_install(IS_TDAQ_INSTALL)
@@ -76,7 +84,7 @@ function(post_build_debug_library name)
   add_custom_command(TARGET ${name}
     POST_BUILD
     COMMAND ${CMAKE_OBJCOPY} --only-keep-debug $<TARGET_FILE:${name}> ${CMAKE_CURRENT_BINARY_DIR}/${our_debug_name}
-    COMMAND ${CMAKE_STRIP} --strip-debug --strip-unneeded $<TARGET_FILE:${name}>
+    COMMAND ${CMAKE_STRIP} ${STRIP_DEBUG_OPTION} ${STRIP_UNNEEDED_OPTION} $<TARGET_FILE:${name}>
     COMMAND ${CMAKE_OBJCOPY} --add-gnu-debuglink=${our_debug_name} $<TARGET_FILE:${name}>
   )
 
@@ -97,7 +105,7 @@ function(post_build_debug_executable name)
   add_custom_command(TARGET ${name}
     POST_BUILD
     COMMAND ${CMAKE_OBJCOPY} --only-keep-debug $<TARGET_FILE:${name}> ${CMAKE_CURRENT_BINARY_DIR}/${our_debug_name}
-    COMMAND ${CMAKE_STRIP} --strip-debug --strip-unneeded $<TARGET_FILE:${name}>
+    COMMAND ${CMAKE_STRIP} ${STRIP_DEBUG_OPTION} ${STRIP_UNNEEDED_OPTION} $<TARGET_FILE:${name}>
     COMMAND ${CMAKE_OBJCOPY} --add-gnu-debuglink=${our_debug_name} $<TARGET_FILE:${name}>
   )
 
-- 
2.45.2

